src/test/java/com/powsybl/openloadflow/network/impl/extensions/SubstationAutomationSystemsTest.java

```diff
@@ -39,13 +47,31 @@ void errorTest() {
         assertEquals("Switch ID to operate is not set", e.getMessage());
     }
 
+    /**
+     * Writes given network to JSON file, then reads the resulting file and returns the resulting network
+     */
+    private static Network jsonWriteAndRead(Network networkInput, ExportOptions options, Path path) {
+        TreeDataFormat previousFormat = options.getFormat();
+        options.setFormat(TreeDataFormat.JSON);
+        Anonymizer anonymizer = NetworkSerDe.write(networkInput, options, path);
+        try (InputStream is = Files.newInputStream(path)) {
+            Network networkOutput = NetworkSerDe.read(is, new ImportOptions().setFormat(TreeDataFormat.JSON), anonymizer);
+            options.setFormat(previousFormat);
+            return networkOutput;
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
     @Test
     void xmlRoundTripTest() throws IOException {
         Network network = AutomationSystemNetworkFactory.create();
 
+        ExportOptions exportOptions = new ExportOptions();
         Network network2 = roundTripXmlTest(network,
-                NetworkSerDe::writeAndValidate,
-                NetworkSerDe::read,
+                (n, p) -> jsonWriteAndRead(n, exportOptions, p),
+                (n, p) -> NetworkSerDe.write(n, exportOptions, p),
+                NetworkSerDe::validateAndRead,
                 "/substationAutomationSystemsRef.xml");
```

Comment:
no need to create your own exportOptions as the default is ok. This simplifies to 
```java
Network network2 = roundTripXmlTest(network,
                this::jsonWriteAndRead,
                NetworkSerDe::write,
                NetworkSerDe::validateAndRead,
```